### DISCUSSION TITLE
Would it be possible to extend the bidirectional_links_generator.rb to allow folders in the path?

### DISCUSSION THREAD

**Discussion by ryan-p-randall** [maximevaillancourt/digital-garden-jekyll-template#109] [Ideas]
I'm really impressed by your template and the plugins you've made. Thank you for all the work you've made public!  

I'm hoping to use just your `bidirectional_links_generator.rb` on my site, which already uses a different theme. Would it be possible to make your links generator work across different folders, in addition to a single folder?  

For instance, instead of `notes/cats.md`, could it be extended to create bidirectional links from `notes/pets/cats.md` and `notes/names/memorable-names.md`? (Tom Critchlow's [folder-based wiki](https://tomcritchlow.com/wiki/) is an example of the structure I have in mind.)

**Comment by maximevaillancourt** [comment]
Hey Ryan, thanks for the kind words!

> Would it be possible to make your links generator work across different folders, in addition to a single folder?

This is already possible. I added an example [here](https://github.com/maximevaillancourt/digital-garden-jekyll-template/commit/680943e70f3c9ea77b69be38311880a5a1f6c804). You can see this on the live demo site [here](https://digital-garden-jekyll-template.netlify.app/your-first-note).

If this doesn't work on your custom Jekyll site, I suspect that you're missing this critical piece in your `_config.yml` file:

https://github.com/maximevaillancourt/digital-garden-jekyll-template/blob/8c72fb4e506faebe72460d37eebc205ad5183af5/_config.yml#L42-L45

Note the double asterisks (`**`) in the path definition: this means that all files within `_notes/`, including subdirectories, with a `.md` extension will be considered as "notes".

**Reply by ryan-p-randall** [reply]
Ah! That part from the `_config.yml` seems to have been the key—it's working now on my site (locally, it's not published yet).

Again, thank you so much for sharing your work **and** your knowledge!
